libres/lib/res_util/memory.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+   Copyright (C) 2011  Equinor ASA, Norway.
+
+   The file 'memory.cpp' is part of ERT - Ensemble based Reservoir Tool.
+
+   ERT is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+   FITNESS FOR A PARTICULAR PURPOSE.
+
+   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+   for more details.
+*/
+
+#include <iostream>
+#include <fstream>
+#include <sys/resource.h>
+
+namespace memory {
+
+/**
+ * Internal util-function
+ */
+inline std::size_t parse_meminfo_linux(const std::string &file,
+                                       const std::string &pattern) {
+    std::ifstream meminfo(file);
+    if (meminfo.is_open()) {
+        std::string tp;
+        std::size_t info;
+        while (std::getline(meminfo, tp)) {
+            if (sscanf(tp.data(), pattern.data(), &info) == 1)
+                return info / 1000;
+        }
+    }
+    return 0;
+}
+
+/**
+ * NOTE: Linux only.
```

Comment:
If this is Linux-only, wrap the whole thing in `#ifdef __linux__` to have the code only be created for Linux. For apple, use `#ifdef __APPLE__`. These are macros defined by the compiler when compiling for the specific operating system.

https://iq.opengenus.org/detect-operating-system-in-c/

Reply:
Good idea! Also makes it simple with a nice message in case not Linux.